client/hooks/use-report-export.ts

```diff
@@ -0,0 +1,202 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useRef, useState } from 'react';
+import apiFetch from '@wordpress/api-fetch';
+import { useDispatch } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+import { recordEvent } from 'wcpay/tracks';
+
+interface ExportResponse {
+	export_id?: string;
+}
+
+interface ExportURLResponse {
+	status: string;
+	download_url?: string;
+}
+
+const maxRetries = 5;
+
+/**
+ * Hook for requesting and polling for a CSV export. E.g. Transactions, Payouts, Disputes.
+ *
+ * @example
+ * const { requestReportExport, isDownloading } = useReportExport();
+ * requestReportExport( {
+ * 	exportRequestURL: '/wc/v3/payments/transactions/download?queryParam=value',
+ * 	exportFileAvailabilityEndpoint: '/wc/v3/payments/transactions/download',
+ * 	userEmail: 'test@example.com',
+ * } );
+ */
+export const useReportExport = () => {
+	const [ isDownloading, setIsDownloading ] = useState( false );
```

Comment:
Does this state reflect "downloading" or that a CSV request is active/processing? 

Might be clearer to rename. Is "in progress" accurate (i.e. keep it simple!)?

Reply:
As the person who wrote this variable name, I totally agree that `isDownloading` is not explicit enough and misleading.

* The actual file download happens instantaneously when the browser triggers the download via the created link element
* Most of the time is spent waiting for the server to generate/prepare the export

Since this is a bit more complex that a simple request, maybe something more descriptive like `isExportInProgress` to better communicate _what_ is in progress – the export operation on server.

Reply:
Changed to `isExportInProgress`